polar-core/src/vm.rs

```diff
@@ -1222,7 +1234,7 @@ impl PolarVirtualMachine {
 
                 // Run compatibility check.
                 self.choose_conditional(
-                    vec![Goal::Run { runnable }],
+                    runnables,
```

Comment:
So this means all LHS must have a compatible isa?

Reply:
Yes, I think that's correct behavior, isn't it? Before it was just trying to pick one according to a complicated set of rules and saying *that* one has to be compatible. This could possibly affect performance but I don't know how likely that actually is

Reply:
Anyway -- the fix actually seems to work without this change, so I'll revert it. But it's obviously not obvious that what we're doing now is correct.